pkg/endpointmanager/manager.go

```diff
@@ -412,23 +413,29 @@ func RegenerateAllEndpoints(owner endpoint.Owner, regenMetadata *endpoint.Extern
 	return &wg
 }
 
-// RegenerateEndpointSetSignalWhenEnqueued regenerates the endpoints represented
-// by endpointIDs. It signals to the provided WaitGroup when all of the endpoints
-// in said set have had regenerations queued up.
-func RegenerateEndpointSetSignalWhenEnqueued(owner endpoint.Owner, regenMetadata *endpoint.ExternalRegenerationMetadata, endpointIDs map[uint16]struct{}, wg *sync.WaitGroup) {
+// RegenerateEndpointSet calls a SetStateLocked for each endpoint from the given
+// set 'epsToRegen' and if it still exists in the list of endpoints manager it
+// regenerates if state transaction is valid. During this process, the endpoint
+// list is locked and cannot be modified.
+// Returns a waiting group that can be used to know when all the endpoints are
+// regenerated.
+func RegenerateEndpointSet(owner endpoint.Owner, regenMetadata *endpoint.ExternalRegenerationMetadata, epsToRegen map[uint16]struct{}) *sync.WaitGroup {
+	mutex.RLock()
```

Comment:
What is the purpose behind locking the endpointmanager here? We are not performing any access on the endpointmanager itself.

Reply:
@ianvernon we are, that's the 1st bug that I'm fixing. See the PR description.

Reply:
Ah, snap, I missed line 429, you are right!